pkg/apis/pipeline/v1beta1/pipeline_validation.go

```diff
@@ -144,6 +144,11 @@ func validatePipelineTask(ctx context.Context, t PipelineTask, taskNames sets.St
 		if t.TaskRef.Kind == "" {
 			errs = errs.Also(apis.ErrInvalidValue("custom task ref must specify kind", "taskRef.kind"))
 		}
+		// Conditions are deprecated so the effort to support them with custom tasks is not justified.
+		// When expressions should be used instead.
+		if len(t.Conditions) > 0 {
+			errs = errs.Also(apis.ErrInvalidValue("custom tasks do not support conditions", "conditions"))
```

Comment:
please refer them to when expressions in the error message as well because that's what they'd see before they look into the docs

Reply:
done